cmd/agent/main.go

```diff
@@ -1,3 +1,171 @@
+	RandomValue   float64
+	PollCount     int64
+}
+
+const DA = "localhost:8080"
+const DPI = 2
+const DRI = 10
+
+type Config struct {
+	Address        string `envDefault:"localhost:8080"`
+	ReportInterval int64  `envDefault:"10"`
+	PollInterval   int64  `envDefault:"2"`
+}
+
+var (
+	address              = flag.String("a", DA, "server adress")
+	pollInterval         = flag.Int("p", DPI, "poll interval in seconds")
+	reportInterval       = flag.Int("r", DRI, "report interval in seconds")
+	PollCount      int64 = 0
+)
+
+func getMetrics(MemStats *runtime.MemStats, PollCount int64) metrics {
+	return metrics{
+		Alloc:         float64(MemStats.Alloc),
+		BuckHashSys:   float64(MemStats.BuckHashSys),
+		Frees:         float64(MemStats.Frees),
+		GCCPUFraction: MemStats.GCCPUFraction,
+		GCSys:         float64(MemStats.GCSys),
+		HeapAlloc:     float64(MemStats.HeapAlloc),
+		HeapIdle:      float64(MemStats.HeapIdle),
+		HeapInuse:     float64(MemStats.HeapInuse),
+		HeapObjects:   float64(MemStats.HeapObjects),
+		HeapReleased:  float64(MemStats.HeapReleased),
+		HeapSys:       float64(MemStats.HeapSys),
+		LastGC:        float64(MemStats.LastGC),
+		Lookups:       float64(MemStats.Lookups),
+		MCacheInuse:   float64(MemStats.MCacheInuse),
+		MCacheSys:     float64(MemStats.MCacheSys),
+		MSpanInuse:    float64(MemStats.MSpanInuse),
+		MSpanSys:      float64(MemStats.MSpanSys),
+		Mallocs:       float64(MemStats.Mallocs),
+		NextGC:        float64(MemStats.NextGC),
+		NumForcedGC:   float64(MemStats.NumForcedGC),
+		NumGC:         float64(MemStats.NumGC),
+		OtherSys:      float64(MemStats.OtherSys),
+		PauseTotalNs:  float64(MemStats.PauseTotalNs),
+		StackInuse:    float64(MemStats.StackInuse),
+		StackSys:      float64(MemStats.StackSys),
+		Sys:           float64(MemStats.Sys),
+		TotalAlloc:    float64(MemStats.TotalAlloc),
+		RandomValue:   rand.Float64(),
+		PollCount:     PollCount,
+	}
+}
+
+func (m *metrics) getUrls() []string {
+	var urls []string
+	v := reflect.ValueOf(*m)
+	for i := 0; i < v.NumField(); i++ {
+		var url string
+		if v.Type().Field(i).Type.String() == "int64" {
+			url = "/update/counter/" + v.Type().Field(i).Name + "/" + fmt.Sprintf("%v", v.Field(i))
+		} else {
+			url = "/update/gauge/" + v.Type().Field(i).Name + "/" + fmt.Sprintf("%v", v.Field(i))
+		}
+		urls = append(urls, url)
+	}
+	return urls
+}
+
+func send(adr *string, urls *[]string) {
+	cl := &http.Client{}
+
+	server := "http://" + *adr
+
+	for _, url := range *urls {
+		//fmt.Println(time.Now().Local().UTC(), "send", url)
+		response, err := cl.Post(server+url, "text/plain", nil)
+		if err != nil {
+			panic(err)
+		}
+		_, err = io.Copy(io.Discard, response.Body)
+		defer response.Body.Close()
+		if err != nil {
+			fmt.Println(err)
+		}
+	}
+
+}
+
+func main() {
+	adr := address
+	var cfg Config
+	opts := env.Options{UseFieldNameByDefault: true}
+	if err := env.ParseWithOptions(&cfg, opts); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
```

Comment:
Можно сделать запись короче:

```suggestion
                log.Fatalf("Error parsing env: %v", err)
```